src/music.cpp

```diff
@@ -531,17 +472,34 @@ uint32_t ff7_music_sound_operation_fix(uint32_t type, uint32_t param1, uint32_t
 	return ret;
 }
 
-void ff7_worldmap_play_custom_battle_music(DWORD* unk1, DWORD* unk2, DWORD* unk3)
+uint32_t ff7_worldmap_music_change(uint32_t type, uint32_t music_id, uint32_t fadetime, uint32_t param3, uint32_t param4, uint32_t param5)
+{
+	if (trace_all || trace_music) trace("%s\n", __func__);
+
+	ff7_last_akao_call_type = type;
+	ff7_last_music_id = music_id;
+
+	if (music_id == 7) { // Battle
+		if (trace_all || trace_music) trace("%s: set music channel to 1\n", __func__);
+		next_music_channel = 1;
+	}
+
+	return ff7_music_sound_operation_fix(type, music_id, fadetime, param3, param4, param5);
+}
+
+void ff7_worldmap_play_custom_battle_music(DWORD* unk1, DWORD* unk2, DWORD* battle_id)
 {
-	ff7_externals.sub_767039(unk1, unk2, unk3);
+	ff7_externals.sub_767039(unk1, unk2, battle_id);
 
 	if (*unk1)
 	{
-		ff7_next_battle_world_id = *unk3;
+		ff7_next_battle_world_id = *battle_id;
 
 		// Now we know the battle scene ID, so we can try to customize battle music in Worldmap too
-		stop_music();
-		ff7_play_midi(7);
+		// Play music
+		((uint32_t(*)(uint32_t, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t))ff7_externals.sound_operation)(

```

Comment:
Oops, removing stop_music() is a bad idea, at least the current music id must be discarded

Reply:
Yes, that's why I used it, in order to flush the current id actually.

Reply:
Done